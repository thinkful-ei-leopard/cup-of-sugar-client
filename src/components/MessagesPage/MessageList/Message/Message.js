import React from 'react'
import styles from './Message.module.scss'
import UserContext from '../../../../contexts/UserContext'
import cx from 'classnames';
import Button from '../../../Button/Button'
import MessagesApiService from '../../../../services/messages-api-service'

export default class Message extends React.Component {

    static contextType=UserContext

    handleDeleteClick = () => {
        MessagesApiService.handleMessageDelete(this.props.message.id)
        this.props.handleMessageDelete(this.props.message.id)
    }

    render() {
        const user = this.context.user
        const currentThread = this.props.currentThread
        let messageHeader = '';
        if (currentThread) {
            if (user.id === this.props.message.user_id) {
                messageHeader = ``
            }

            else if (user.id !== this.props.message.user_id) {
                if(user.id === currentThread.user_id1) {
                    messageHeader = `--${currentThread.name2} (${currentThread.user_name2})`
                }
                else if(user.id === currentThread.user_id2) {
                    messageHeader = `--${currentThread.name1} (${currentThread.user_name1})`
                }
            }
        }
        
        return (
            <li className={cx(styles.messageLi, user.id === this.props.message.user_id ? styles.sentMessage : styles.receivedMessage)}>
                <h3>{messageHeader}</h3>
                <p className={styles.messageContent}>{this.props.message.content}</p>
                {/* <button type='delete' className={styles.deleteMessageButton} onClick={(e) => {
                    e.preventDefault()
                    this.handleDeleteClick()
                }}>X</button> */}
            </li>
        )
    }
}