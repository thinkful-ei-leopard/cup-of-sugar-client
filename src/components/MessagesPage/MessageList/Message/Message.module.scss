@import '../../../../styles/_variables';
@import '../../../../styles/breakpoints';

.messageLi {
    margin-top: 1rem;
}