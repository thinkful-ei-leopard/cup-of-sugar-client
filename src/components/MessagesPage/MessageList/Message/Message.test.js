import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Message from './Message';

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
    <Router>
      <Message
        currentThread={null}
        handleMessageDelete={() => {}}
        message={{}}
      />
    </Router>,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <Router>
        <Message
          currentThread={null}
          handleMessageDelete={() => {}}
          message={{}}
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
