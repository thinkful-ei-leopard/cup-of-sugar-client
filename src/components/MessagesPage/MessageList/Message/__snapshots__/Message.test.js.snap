// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<li
  className="messageLi sentMessage"
>
  <p>
    No messages with this user yet!
  </p>
</li>
`;
